File: src/test/java/messages/RecordTest.java
package messages;

import org.junit.Before;
import org.junit.Test;

import java.nio.ByteBuffer;

import static org.junit.Assert.assertEquals;

/**
 * @author ilya2
 *         created on 29.03.2017
 */
public class RecordTest {
    private Record testRecord;

    @Before
    public void setTestMessage() {
        testRecord = new Record("name", "header", "message");
    }

    @Test
    public void toAndFromByteBuffer() throws Exception {
        ByteBuffer buffer = testRecord.toByteBuffer();
        Record fromBB = Record.fromByteBuffer(buffer);
        assertEquals(testRecord, fromBB);
    }
}

File: src/test/java/messages/RecordTest.kt
package messages

import org.junit.Before
import org.junit.Test

import org.junit.Assert.assertEquals

/**
 * @author ilya2
 * *         created on 29.03.2017
 */
class RecordTest {
    private var testRecord: Record? = null

    @Before
    fun setTestMessage() {
        testRecord = Record("name", "header", "message")
    }

    @Test
    fun toAndFromByteBuffer() {
        val buffer = testRecord!!.toByteArray()
        val fromBB = Record.fromInputStream(buffer.inputStream())
        assertEquals(testRecord, fromBB)
    }
}